apps/omg_watcher/lib/exit_processor/core.ex

```diff
@@ -507,6 +587,190 @@ defmodule OMG.Watcher.ExitProcessor.Core do
     {chain_validity, events}
   end
 
+  @spec get_output_challenge_data(ExitProcessor.Request.t(), t(), Signed.tx_bytes(), 4..7) :: %{
+          in_flight_txbytes: Transaction.tx_bytes(),
+          in_flight_output_pos: Utxo.Position.t(),
+          in_flight_tx_inclusion_proof: <<_::_*256>>,
+          spending_tx_bytes: Transaction.tx_bytes(),
+          spending_input_index: 0..3,
+          spending_sig: <<_::520>>
+        }
+  def get_output_challenge_data(
+        %ExitProcessor.Request{blocks_result: _blocks, piggybacked_blocks_result: _piggybacked_blocks} = _request,
+        %__MODULE__{in_flight_exits: _ifes} = _state,
+        _txbytes,
+        _output_index
+      ) do
+    #   "in_flight_txbytes": "F3170101C0940000...",
+    #   "in_flight_output_pos": 21000634002,
+    #   "in_flight_proof": "03F451067A805540000...",
+    #   "spending_txbytes": "F847010180808080940000...",
+    #   "spending_input_index": 1,
+    #   "spending_sig": "9A23010180808080940000..."
+  end
+
+  @spec get_input_challenge_data(ExitProcessor.Request.t(), t(), Signed.tx_bytes(), 0..7) ::
+          {:ok, input_challenge_data() | output_challenge_data()} | {:error, atom()}
+  def get_input_challenge_data(
+        request,
+        %__MODULE__{in_flight_exits: ifes} = state,
+        txbytes,
+        input_index
+      ) do
+    with {:ok, tx} <- Transaction.decode(txbytes),
+         {:known_ife, true} <- {:known_ife, Transaction.hash(tx) in Map.keys(ifes)},
```

Comment:
follow the conventions that are already employed here (see `find_competitor` for example), so:

```
with ....
      {:ok, something} <- find_something_specific(...),
```

where `find_something_specific` returns `{:error, specific_testable_reason}` in case of a critical failure 

OR

```
with ....
      true <- specific_condition_met?()
```

where `specific_condition_met` returns either `true` or `{:error, specific_testable_reason}`

Reply:
done

Reply:
see here